libraries/libfc/include/fc/exception/exception.hpp

```diff
@@ -496,6 +496,22 @@ namespace fc
       wlog( "${details}", ("details",e.to_detail_string()) ); \
    }
 
+#define FC_LOG_AND_DROP_NO_INTERPROCESS_BAD_ALLOC( ... )  \
+   catch( fc::exception& er ) { \
+      wlog( "${details}", ("details",er.to_detail_string()) ); \
+   } catch( const std::exception& e ) {  \
+      fc::std_exception_wrapper sew( \
+                FC_LOG_MESSAGE( warn, "rethrow ${what}: ",FC_FORMAT_ARG_PARAMS( __VA_ARGS__  )("what",e.what()) ), \
+                std::current_exception(), \
+                BOOST_CORE_TYPEID(e).name(), \
+                e.what() ) ; \
+      wlog( "${details}", ("details",sew.to_detail_string()) ); \
+   } catch( ... ) {  \
+      fc::unhandled_exception e( \
+                FC_LOG_MESSAGE( warn, "rethrow", FC_FORMAT_ARG_PARAMS( __VA_ARGS__) ), \
+                std::current_exception() ); \
+      wlog( "${details}", ("details",e.to_detail_string()) ); \
+   }
```

Comment:
I think this could just be what `FC_LOG_AND_DROP` becomes. Just remove the `catch` for `bad_alloc` from `FC_LOG_AND_DROP`. All it does is `throw`, so they are equivalent.

Reply:
I was worried removing `bad_alloc` from `FC_LOG_AND_DROP` could affect other users of `FC_LOG_AND_DROP`. You are correct. After `bad_alloc` catch is removed, an exception of `bad_alloc` will be caught by the final `catch ( ... )`, logged, and re-thrown. Old and new versions are equivalent. Thanks.